### PR TITLE
Added Macro Method chaining

### PR DESCRIPTION
**Pull request by o-reo**
Added a method chaining behavior to Macros

Example: Gets the workspace with the specified name and sets its id in config, then dumps command details in a json file
```python
from Babylon.utils import Macro

Macro("my macro") \
    .step(["powerbi", "workspace", "get", "-n", "my_workspace"], store_at="workspace") \
    .step(["config", "deployment", "set-variable", "workspace_id", "%datastore%workspace.data.id")  \
    .dump("set_workspace.json")
```

### CODE REVIEW

File: Babylon/utils/macro.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    def __init__(self, name: str):
+        self.name = name
+        self._responses: list[CommandResponse] = []
+        self.env = Environment()
+        self._status = self.STATUS_OK
+
+    def step(self,
+             command_line: list[str],
+             optional: bool = False,
```
Comment (enocquet) [inline Babylon/utils/macro.py:30]: When possible, the naming of boolean vars should be in the form "is_..." so her we can imagine is_option maybe ?

File: Babylon/utils/macro.py [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+             store_at: Optional[str] = None,
+             run_if: bool = True) -> "Macro":
```
Comment (enocquet) [inline Babylon/utils/macro.py:32]: Same comment here: it is not easy to understand what the parameter "run_if" does exactly at the first sight...

File: Babylon/groups/webapp/commands/deploy.py [hunk 3]
```diff
@@ -17,91 +15,42 @@
...
-    logger.info("10 - Adding powerbi credentials in Static WebApp settings...")
-    r_swa_stgs = run_command(["azure", "staticwebapp", "app-settings", "update", f"Azure{deployment_name}WebApp"])
-    if r_swa_stgs.has_failed():
-        logger.warning("Failed to update static webapp settings")
-    logger.info("WebApp deployment was successfull, please wait for workflow to finish")
-    logger.info(f'WebApp url: https://{r_swa.data.get("properties", {}).get("defaultHostname", "")}')
-    return CommandResponse.success()
+    Macro("webapp deploy") \
+        .step(["azure", "staticwebapp", "create", f"Azure{deployment_name}WebApp"]) \
+        .wait(5) \
```
[the remark this answers is not in the document]
Comment (o-reo) [reply Babylon/groups/webapp/commands/deploy.py:28]: I need some feedback from CAT to see if it is enough for them. I would have run a get request to check if the resource has been created. It implies a bit of additional work so I don't want to rush on it if it is not needed.

File: Babylon/utils/macro.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        sleep(delay)
+        return self
+
+    def dump(self, output_file: str) -> "Macro":
+        """Dump command responses data in a json file"""
+        compiled = [{"header": self.name}]
+        compiled.extend([response.to_dict() for response in self._responses])
+        path = Path(output_file)
+        output_path = path
+        for idx in range(1, 10000):
```
[the remark this answers is not in the document]
Comment (o-reo) [reply Babylon/utils/macro.py:85]: I did not want to put an infinite loop, but I can put a `while True`

File: Babylon/utils/macro.py (hunk 1 above)
Comment (o-reo) [reply Babylon/utils/macro.py:30]: `is_required`

File: Babylon/utils/macro.py (hunk 2 above)
Comment (o-reo) [reply Babylon/utils/macro.py:32]: Oh, I thought it was alright. I will document in docstring it as I don't know how to name it better.

File: Babylon/groups/webapp/commands/deploy.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (enocquet) [reply Babylon/groups/webapp/commands/deploy.py:28]: Ok then we could go for some polling, so let's keep this in mind for upcoming work if this configuration doesn't break on users for now :)

[the remark this answers is not in the document]
Comment (o-reo) [reply Babylon/groups/webapp/commands/deploy.py:28]: I'll see on monday if I have an idea to do it right

File: Babylon/utils/macro.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (enocquet) [reply Babylon/utils/macro.py]: The while true loop doesn't look nice to me... I've made a quick possible solution for this issue. To make things simple, once your PR is merged, I'll submit you this fix

**Comment by enocquet** [comment]
approved

File: Babylon/utils/macro.py [hunk 5]
```diff
@@ -91,12 +91,10 @@ def dump(self, output_file: str) -> "Macro":
...
         # Get an available filename of the form output_file_xxxx.json
-        idx = 0
-        while True:
-            if not output_path.exists():
-                break
-            idx += 1
+        idx = 1
+        while output_path.exists():
             output_path = path.with_stem(f"{path.stem}_{idx}")
+            idx += 1
```
Comment (o-reo) [inline Babylon/utils/macro.py:97]: better @enocquet ?

Comment (enocquet) [reply Babylon/utils/macro.py]: yes

**Pull request merged**